src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs

```diff
@@ -254,6 +254,11 @@ private void WaitUntilSiteStarted()
                 }
 
                 HostProcess = Process.GetProcessById(workerProcess.ProcessId);
+
+                // Ensure w3wp.exe is killed if test process termination is non-graceful.
+                // Prevents locked files when stop debugging unit test.
+                ProcessTracker.Add(HostProcess);
```

Comment:
How can this work for w3wp? The worker process would be immediately recycled and the dll would be locked again when killed.

Reply:
When the test process exits, `w3wp.exe` (and the `dotnet.exe` child process of `w3wp.exe`) are killed.  It's true that `w3wp.exe` is immediately restarted, but the `dotnet.exe` child process will not.  So, while some dlls may still be locked, it seems better than doing nothing.

Reply:
The file that is locked is always aspnetcore[v2].dll, so this really doesn't help at all.

Reply:
I would argue its still slightly better to kill `dotnet.exe` than do nothing.  However, I'm also fine limiting this change to IIS Express, if we think it's misleading or noisy to try killing `w3wp.exe`.

Reply:
Anyway to register a callback? Doesn't look like it, but maybe we can add a callback to undo the redirection.config changes on exit.

Reply:
I believe the best we can do with Windows Job Objects is to kill processes.  If we wanted to undo the `redirection.config` changes (or anything else), we could create a watchdog process which detects when the test process terminates (e.g. via anonymous pipes) and then performs cleanup.